### PR TITLE
Add missing features to ingress metrics

### PR DESCRIPTION
**Pull request by spencerhance**
These features have been implemented but are currently not included in the metrics
- SSLPolicy
- Regional Static IP
- Custom Health Checks

This PR adds metrics for them
All of them can be cherry-picked to v1.9 + v1.10

HTTPS Redirects will be added in a follow up since it is currently 1.10 only

### CODE REVIEW

**Comment by spencerhance** [comment]
/assign @prameshj 

Pavithra, any chance you have bandwidth to review this?

**Comment by prameshj** [comment]
Would it be better to split into 2 PRs - one with metrics that can be cherrypicked to 1.9 and one with those that cannot?

File: pkg/metrics/metrics_test.go [hunk 1]
```diff
@@ -484,8 +500,8 @@ var (
 					Namespace: defaultNamespace,
 					Name:      "ingress13",
 					Annotations: map[string]string{
-						StaticIPNameKey: "user-spec-static-ip",
-						staticIPKey:     "user-spec-static-ip",
+						StaticGlobalIPNameKey: "user-spec-static-ip",
+						staticIPKey:           "user-spec-static-ip",
```
Comment (prameshj) [inline pkg/metrics/metrics_test.go:504]: nit: should this value be an ip address?

Comment (spencerhance) [reply pkg/metrics/metrics_test.go:504]: Both of the annotations represent the name of the address, rather than the address itself

**Comment by spencerhance** [comment]
@prameshj Good call!  I removed redirects from this.  Once it is merged I'll submit a follow up PR.

File: pkg/metrics/features.go [hunk 2]
```diff
@@ -191,15 +201,24 @@ func featuresForIngress(ing *v1beta1.Ingress) []feature {
 	if val, ok := ingAnnotations[staticIPKey]; ok && val != "" {
```
Comment (prameshj) [inline pkg/metrics/features.go:201]: Does this mean user has to specify both staticIPKey and StaticGlobalIPNameKey in order to use a given ip address name? Why do we need 2?

Comment (spencerhance) [reply pkg/metrics/features.go:201]: oh, no staticIPKey is used as a status annotation

File: pkg/metrics/features.go [hunk 3]
```diff
@@ -191,15 +201,28 @@ func featuresForIngress(ing *v1beta1.Ingress) []feature {
...
 			klog.V(6).Infof("User specified static IP for ingress %s: %s", ingKey, val)
 			features = append(features, specifiedStaticGlobalIP)
 		} else {
 			features = append(features, managedStaticGlobalIP)
 		}
 	}
+
+	// Check for regional static IP
+	// We do this separately from the global static IP because regional static IP
+	// does not support the StaticIPKey annotation or a managed regional IP
```
Comment (prameshj) [inline pkg/metrics/features.go:215]: Can you rephrase this to say - "Controller does not populate StaticIPKey annotation when processing Regional static IP"?

Also, these annotations are also defined in annotations/ingress.go. Do they need to be redefined in features.go?

Comment (spencerhance) [reply pkg/metrics/features.go:215]: @skmatti what was the reason for this?  Was there a circular dependency?

Comment (spencerhance) [reply pkg/metrics/features.go]: Updated the comment

Comment (skmatti) [reply pkg/metrics/features.go]: This was to enable backporting for the metrics PR to the release branches that do not have the annotations defined. Maybe it does not apply for these metrics.

Comment (spencerhance) [reply pkg/metrics/features.go]: Okay I removed the annotation

File: pkg/metrics/features.go [hunk 4]
```diff
@@ -191,15 +199,27 @@ func featuresForIngress(ing *v1beta1.Ingress) []feature {
...
 			klog.V(6).Infof("User specified static IP for ingress %s: %s", ingKey, val)
 			features = append(features, specifiedStaticGlobalIP)
 		} else {
 			features = append(features, managedStaticGlobalIP)
 		}
 	}
+
+	// Check for regional static IP
+	// We do this separately from the global static IP because Controller does not
+	// populate StaticIPKey annotation when processing Regional static IP regional static IP
```
Comment (prameshj) [inline pkg/metrics/features.go:213]: remove

Comment (spencerhance) [reply pkg/metrics/features.go]: done
